packages/types/src/metrics.ts

```diff
@@ -5,27 +5,21 @@ import type { Primitive } from './misc';
  * An abstract definition of the minimum required API
  * for a metric instance.
  */
-export abstract class MetricInstance {
+export interface MetricInstance {
   /**
    * Returns the weight of the metric.
    */
-  public get weight(): number {
-    return 1;
-  }
+  weight: number;
```

Comment:
This now becomes a variable, not a `getter`

Reply:
Typescript-wise, this is the same. Public API wise too!